### PR TITLE
Nikiforov Oleksandr java hw2

### PR DESCRIPTION
**Pull request by olnikiforov**

### CODE REVIEW

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture1/tasks/_1/Task1.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.zoolatech.lecture1.tasks._1;
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture1/tasks/_1/Task1.java:1]: Please remove files from the PR that are not related to the second homework and were already defined here: https://github.com/zoolacode/zoola-academy-homework/pull/91/files

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_1/Calculcator.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    public float getValue() {
+        return value;
+    }
+
+    public void setValue(float value) {
+        this.value = value;
+    }
+
+    public float multiplication(float variable){
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_1/Calculcator.java:18]: `variable` might be not the best naming for the method argument, try to think about a better one

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_1/Task1.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                    break;
+                case '*':
+                    System.out.println("Value to calculate: ");
+                    variable = sc.nextFloat();
+                    result = calculcator.multiplication(variable);
+                    System.out.println("Result = " + result);
+                    break;
+                case '%':
+                    System.out.println("Bue!");
+                    break end;
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_1/Task1.java:55]: instead of a labeled break you could also use `return` without arguments or the do-while loop with the exit condition `symbol = '%' `

Comment (tromanyuk) [reply java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_1/Task1.java:55]: One more idea: you can split the accepting input for the operation and checking for the exit symbol into two operations: this would allow to remove the common code from case blocks.
e.g.,
```
variable = sc.nextFloat();
shouldExit = sc.next();
```

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Task2.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ * and direction as the initial robot location. Robot must not be able to get outside of a room - in such case the robot should move as much as possible
+ * to the room border, print a warning message and stop. All next moves in the same direction should be prevented.
+ */
+
+
+public class Task2 {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Input max range of the room: ");
+        int n = sc.nextInt();
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Task2.java:19]: the robot needs to accept two parameters: width and height of the room

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Task2.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        robotWrok(n);
+    }
+
+
+    public static void robotWrok(int n){
+        Scanner sc = new Scanner(System.in);
+        System.out.println("DO you want to enter the starting coordinates manually print 1, no - 2: ");
+        int tmp = sc.nextInt();
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Task2.java:27]: `tmp` is not the best naming for this variable

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    public void setH(int h) {
+        this.h = h;
+    }
+
+    public int getH() {
+        return h;
+    }
+
+    public void getCoordinates(){
+        System.out.println("Current coordinates are ("+getW()+";"+getH()+")");
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:89]: since it's the instance method it can directly access these values without a need to call methods

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Task2.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                    System.out.println("Input L to move left, input R to move right:");
+                    robot.changeDirection(sc.next().charAt(0));
+                    robot.getDirection();
+                    break;
+                case 3:
+                    System.out.println();
+                    robot.getCoordinates();
+                    robot.getDirection();
+                    break;
+                case 0:
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Task2.java:58]: similar to the previous task

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
+package com.zoolatech.lecture2.tasks._2;
+
+public class Robot {
+    private int w, h, n;
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:4]: `width` and `height` might be more meaningful names for these variables

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Task2.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+public class Task2 {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Input max range of the room: ");
+        int n = sc.nextInt();
+        robotWrok(n);
+    }
+
+
+    public static void robotWrok(int n){
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Task2.java:24]: `work`? 😄

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+    private int direction;
+
+    public Robot(int n) {
+        this.direction =0;
+        this.w = 0;
+        this.h = 0;
+        this.n = n-1;
+    }
+
+    public void move(int step){
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:18]: please also add a method that always moves a robot by 1 tile

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+            } else if (this.h<1) {
+                System.out.println("The robot tried to get out of the room.");
+                this.h = 1;
+            }
+        }
+
+    }
+    public void changeDirection(char direct){
+        if (direct=='L'){
+            this.direction+=1;
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:60]: nice idea, but increment/decrement operators might be more readable

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+        } else if (direct=='R') {
+            this.direction-=1;
+        }else{
+            System.out.println("Incorrect input!");
+        }
+        this.direction = this.direction % 4;
+        if (this.direction<0){
+            this.direction+=4;
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:68]: you can just assign the max value

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+    public void getDirection(){
+        switch (direction){
+            case 0:
+                System.out.println("The current direction of the robot to the right");
+                break;
+            case 1:
+                System.out.println("The current direction of the robot upwards");
+                break;
+            case 2:
+                System.out.println("The current direction of the robot to the left");
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:101]: break statements are missing for case = 2 and 3

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+
+    public void move(int step){
+        if (this.direction==0){
+            this.w+=step;
+            if(this.w>this.n){
+                System.out.println("The robot tried to get out of the room.");
+                this.w = this.n;
+            }else if (this.w<1) {
+                System.out.println("The robot tried to get out of the room.");
+                this.w = 1;
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:26]:
```
The north-west corner of a room is represented by coordinates (0,0); north-east - (W-1, 0); south-east - (W-1, H-1); south-west - (0, H-1)
```
zero is a valid value, but `n` - is not

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 15]
```diff
@@ -0,0 +1,109 @@
+package com.zoolatech.lecture2.tasks._2;
+
+public class Robot {
+    private int w, h, n;
+    // direction = 0 - move right
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:5]: the code might be more readable if these values are defined as constants

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+    public Robot(int n) {
+        this.direction =0;
+        this.w = 0;
+        this.h = 0;
+        this.n = n-1;
+    }
+
+    public void move(int step){
+        if (this.direction==0){
+            this.w+=step;
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:20]: We add a positive value - why do we need to have an else block in the if-else block? It's enough to check only if a current coordinate is bigger than a room boundary. Similar for other directions.

**Comment by tromanyuk** [comment]
Nice work, please resolve issues in the comments

File: java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java [hunk 17]
```diff
@@ -0,0 +1,129 @@
...
+    public Robot(int maxWidth, int maxHeight) {
+        this.direction = 0;
+        this.width = 0;
+        this.height = 0;
+        this.maxWidth = maxWidth - 1;
+        this.maxHeight = maxHeight - 1;
+    }
+
+    public void moveOnTile() {
+        if (this.direction == 0) {
```
Comment (tromanyuk) [inline java/Nikiforov-Oleksandr/src/com/zoolatech/lecture2/tasks/_2/Robot.java:24]: all of the code for this method could simply be replaced with `move(1)`

**Review by tromanyuk — DISMISSED** [review]

**Comment by tromanyuk** [comment]
There is a merge conflict which you need to resolve before I can merge your changes

**Comment by olnikiforov** [comment]
I can't resolve a conflict, because I don't have rights. The conflict appeared because I did not see that the commits were added from the 1st Hw, and I deleted the files, after your comment, what to do?)

**Comment by tromanyuk** [comment]
Please merge changes from a master branch and keep files from the first homework during the resolution of the merge conflict.

**Comment by olnikiforov** [comment]
Merge conflict resolved

**Review by tromanyuk — APPROVED** [review]
